src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -142,15 +141,35 @@ public PrivateKey getCAPrivateKey() {
      * @throws KeyStoreException if unable to retrieve the certificate
      */
     public X509Certificate getCACertificate() throws KeyStoreException {
-        return (X509Certificate) getCaCertificateChain()[0];
+        X509Certificate[] certChain = getCaCertificateChain();
+        if (certChain == null) {
+            throw new KeyStoreException("No CA certificate configured");
+        }
+        return certChain[0];
+    }
+
+    private void setCaCertificateChain(Certificate... caCertificateChain) throws KeyStoreException {
+        if (caCertificateChain == null) {
+            throw new KeyStoreException("No certificate chain provided");
+        }
+        for (Certificate cert : caCertificateChain) {
+            if (!(cert instanceof X509Certificate)) {
+                throw new KeyStoreException("Unsupported certificate type");
+            }
+        }
+        setCaCertificateChain(Arrays.stream(caCertificateChain).toArray(X509Certificate[]::new));
```

Comment:
caCertificateChain is already an array, `...` is an array. Why not just cast it?

Reply:
Class cast exception. The objects are x509 certs but the array is not.